docs/querying/datasource.md

```diff
@@ -302,12 +302,15 @@ use join datasources.
 SQL joins take the form:
 
 ```
-<o1> [ INNER | LEFT [OUTER] ] JOIN <o2> ON <condition>
+<o1> [ INNER | LEFT [OUTER] | CROSS ] JOIN <o2> ON <condition>
 ```
 
-The condition must involve only equalities, but functions are okay, and there can be multiple equalities ANDed together.
-Conditions like `t1.x = t2.x`, or `LOWER(t1.x) = t2.x`, or `t1.x = t2.x AND t1.y = t2.y` can all be handled. Conditions
-like `t1.x <> t2.x` cannot currently be handled.
+JOIN conditions have the following requirements:
+- You can only use equality statements in JOIN conditions. For example: `t1.x = t2.x`
+- You can use functions on the operands in JOIN conditions. For example: `LOWER(t1.x) = t2.x`
+- You can apply multiple equality conditions using the AND operator. For example: `t1.x = t2.x AND t1.y = t2.y`
+- Inequality conditions are unsupported. For example: `t1.x <> t2.x`
+- Conditions are optional for CROSS JOIN.
```

Comment:
I don't think CROSS JOIN will accept join conditions ...

Reply:
From a comment in `Parser.jj`:
> We allow CROSS JOIN to have a join condition, even though that is not valid SQL; the validator will catch it.

This seems to say that the Calcite parser allows join conditions, but the Calcite validator fails if a condition is provided.

To avoid confusing users with this implementation detail, perhaps say:

> [ [ INNER | LEFT [OUTER] JOIN <o2> ON <condition> ] | CROSS JOIN <o2>

Or, just address it in text:

> CROSS JOIN does not accept an `ON` condition.

Reply:
```suggestion
- CROSS JOIN does not accept an ON condition.
```
@paul-rogers I did this in both places because I the syntax should be correct, but we can state it in English here for ease/clarity